go/chat/devstorage.go

```diff
@@ -0,0 +1,102 @@
+package chat
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/keybase/client/go/chat/globals"
+	"github.com/keybase/client/go/chat/utils"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+type DevConversationBackedStorage struct {
+	globals.Contextified
+	utils.DebugLabeler
+
+	ri func() chat1.RemoteInterface
+}
+
+func NewDevConversationBackedStorage(g *globals.Context, ri func() chat1.RemoteInterface) *DevConversationBackedStorage {
+	return &DevConversationBackedStorage{
+		Contextified: globals.NewContextified(g),
+		DebugLabeler: utils.NewDebugLabeler(g.GetLog(), "DevConversationBackedStorage", false),
+		ri:           ri,
+	}
+}
+
+func (s *DevConversationBackedStorage) Put(ctx context.Context, name string, src interface{}) (err error) {
+	defer s.Trace(ctx, func() error { return err }, "Put(%s)", name)()
+	if s.G().ActiveDevice == nil {
+		return libkb.LoggedInError{}
+	}
+	dat, err := json.Marshal(src)
+	if err != nil {
+		return err
+	}
+	uid := gregor1.UID(s.G().ActiveDevice.UID().ToBytes())
+	username := s.G().ActiveDevice.Username(libkb.NewMetaContext(ctx, s.G().ExternalG())).String()
+	conv, err := NewConversation(ctx, s.G(), uid, username, &name, chat1.TopicType_DEV,
+		chat1.ConversationMembersType_IMPTEAMNATIVE, keybase1.TLFVisibility_PRIVATE, s.ri)
+	if err != nil {
+		return err
+	}
+	if _, _, err = NewBlockingSender(s.G(), NewBoxer(s.G()), s.ri).Send(ctx, conv.GetConvID(),
+		chat1.MessagePlaintext{
+			ClientHeader: chat1.MessageClientHeader{
+				Conv:        conv.Info.Triple,
+				TlfName:     username,
+				MessageType: chat1.MessageType_TEXT,
+			},
+			MessageBody: chat1.NewMessageBodyWithText(chat1.MessageText{
+				Body: string(dat),
+			}),
+		}, 0, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *DevConversationBackedStorage) Get(ctx context.Context, name string, dest interface{}) (found bool, err error) {
+	defer s.Trace(ctx, func() error { return err }, "Get(%s)", name)()
+	if s.G().ActiveDevice == nil {
```

Comment:
can you move `assertLoggedInUID` from the server to `utils` so we can use it throughout the chat package? that checks `ActiveDevice.HaveKeys()` and we should keep it consistent (i don't think there is a practical difference but we do this check everywhere so it's confusing if it's different)

Reply:
Sure